### PR TITLE
core(axe): use our html snippet rather than axe's

### PR DESCRIPTION
**Pull request by paulirish**
found while working with @gMakunde on his audit. the autofilltypepredictions that are in the DOM were picked up by axe as the `node.html`.. it's just a basic outerHTML grab: https://github.com/dequelabs/axe-core/blob/0cd1eade459a5e3b36220ab9a4d3884db678cc99/lib/core/utils/dq-element.js#L16-L22

since our getOuterHTMLSnippet() is far more sophisticated these days.. we should prefer that.

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ const expectations = [
                 node: {
                   'type': 'node',
                   'selector': '#aria-required-children',
-                  'snippet': '<div id="aria-required-children" role="radiogroup">\n        <div></div>\n      </div>',
```
Comment (paulirish) [inline lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js:92]: this one seems good.

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ const expectations = [
                     'height': '>0',
                   },
                   'selector': '#aria-hidden-focus',
-                  'snippet': '<div id="aria-hidden-focus" aria-hidden="true">\n        <button>Focusable Button</button>\n      </div>',
+                  'snippet': '<div id="aria-hidden-focus" aria-hidden="true">',
```
Comment (paulirish) [inline lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js:56]: at first blush we're losing 'Focusable Button' but we actually have that in `nodeLabel`. so we good.

same with a few other of these cases.

File: lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js [hunk 3]
```diff
@@ -363,7 +363,7 @@ const expectations = [
                   'type': 'node',
                   'selector': 'h3',
                   'path': '2,HTML,1,BODY,37,SECTION,1,H3',
-                  'snippet': '<h3>sub-sub-header</h3>',
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js:366]: dont like. should we change our snippet?

Comment (paulirish) [reply lighthouse-cli/test/smokehouse/test-definitions/a11y/expectations.js:366]: nodeLabel has it covered.

**Pull request merged**